src/resourcemanager/resourcemanager.cpp

```diff
@@ -0,0 +1,490 @@
+        Poco::JSON::Object object;
+
+        object.set("name", resource.mName.CStr());
+        object.set("groups", ToJSONArray(resource.mGroups));
+        object.set("mounts", DumpMounts(resource.mMounts));
+        object.set("env", ToJSONArray(resource.mEnv));
+        object.set("hosts", DumpHosts(resource.mHosts));
+
+        array.add(object);
+    }
+
+    return array;
+}
+
+/***********************************************************************************************************************
+ * JSONProvider
+ **********************************************************************************************************************/
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+Error JSONProvider::DumpNodeConfig(const sm::resourcemanager::NodeConfig& config, String& json) const
+{
+    try {
+        Poco::JSON::Object object;
+
+        object.set("version", config.mVersion.CStr());
+        object.set("nodeType", config.mNodeConfig.mNodeType.CStr());
+        object.set("priority", config.mNodeConfig.mPriority);
+        object.set("devices", DumpDevices(config.mNodeConfig.mDevices));
+        object.set("resources", DumpResources(config.mNodeConfig.mResources));
+        object.set("labels", ToJSONArray(config.mNodeConfig.mLabels));
+
+        std::ostringstream ss;
+        Poco::JSON::Stringifier::stringify(object, ss);
+
+        json = ss.str().c_str();
+    } catch (const Poco::Exception& e) {
+        LOG_ERR() << "Error dumping config: " << e.what();
+
+        return Error(AOS_ERROR_WRAP(ErrorEnum::eFailed), e.what());
+    }
+
+    return ErrorEnum::eNone;
+}
+
+Error JSONProvider::ParseNodeConfig(const String& json, sm::resourcemanager::NodeConfig& config) const
+{
+    try {
+        Poco::JSON::Parser           parser;
+        auto                         result = parser.parse(json.CStr());
+        CaseInsensitiveObjectWrapper object(result.extract<Poco::JSON::Object::Ptr>());
+
+        config.mVersion              = object.GetValue<std::string>("version").c_str();
+        config.mNodeConfig.mNodeType = object.GetValue<std::string>("nodeType").c_str();
+        config.mNodeConfig.mPriority = object.GetValue<uint32_t>("priority");
+
+        ParseDevices(object, config.mNodeConfig.mDevices);
+        ParseResources(object, config.mNodeConfig.mResources);
+        ParseLabels(object, config.mNodeConfig.mLabels);
+    } catch (const common::utils::AosException& e) {
+        LOG_ERR() << "Error parsing config: " << e.message().c_str();
+
+        return Error(e.GetError(), e.message().c_str());
+    } catch (const Poco::Exception& e) {
+        LOG_ERR() << "Error parsing config: " << e.what();
+
+        return Error(AOS_ERROR_WRAP(ErrorEnum::eFailed), e.what());
+    }
+
+    return ErrorEnum::eNone;
+}
```

Comment:
just propese, maybe move above function to file like "utils.cpp". Up to you

Reply:
I'd keep it here and more to  utils later if needed by some other module